upgrade_common_services.sh

```diff
@@ -148,6 +150,101 @@ function check_preqreqs() {
     fi
 }
 
+function pre_zen(){
+    local csNS=$1
+
+    STEP=$((STEP + 1 ))
+    msg ""
+    title "[${STEP}] Detecting Operator Condition of zen operator v1.4.2 in ${csNS} namespace..."
+    msg "-----------------------------------------------------------------------"
+
+    if oc get operatorcondition -n ${csNS} 2>/dev/null | grep ibm-zen-operator ; then
+        msg "Removing Operator Condition of zen operator v1.4.2..."
+        list=$(oc get operatorcondition -o custom-columns=operatorcondition:.metadata.name --no-headers -n ${csNS} | grep ibm-zen-operator 2>/dev/null)
+        for name in ${list};do
+            if [ ! -z "${name}" ] && [[ "${name}" =~ ibm-zen-operator.v1.4.2 ]]; then
+                oc delete operatorcondition ${name} -n ${csNS} || true
+            fi
+        done
+    else
+        msg "Operator Condition in namespace ${csNS} not found, skipping..."
+    fi
+}
+
+function zenopr_check() {
+    local csNS=$1
+
+    STEP=$((STEP + 1 ))
+    msg ""
+    title "[${STEP}] Checking if Zen Operator has been upgraded to the middle version..."
+    msg "-----------------------------------------------------------------------"
+
+    sleep 60
+    while true; do
+        # check if installedCSV is the same as currentCSV
+        installedCSV=$(oc get subscription.operators.coreos.com ibm-zen-operator -n ${csNS} --ignore-not-found -o jsonpath={.status.installedCSV})
+        currentCSV=$(oc get subscription.operators.coreos.com ibm-zen-operator -n ${csNS} --ignore-not-found -o jsonpath={.status.currentCSV})
+
+        if [[ $installedCSV != $currentCSV ]]; then
+            install_plan=$(oc get subscription.operators.coreos.com ibm-zen-operator -n ${csNS} --ignore-not-found -o jsonpath={.spec.installPlanApproval})
+            echo "install plan" $install_plan
+            if [[ $install_plan == "Manual" ]]; then
+                error "install plan is on Manual mode, need to approve it manually to upgrade Zen operator"
+            fi
+            warning "install plan is on Automatic mode, waiting for installedCSV ${installedCSV} upgrade to currentCSV ${currentCSV}..."
+            sleep 5
+        else
+            success "installedCSV ${installedCSV} is the same as currentCSV ${currentCSV}."
+            break
+        fi
+    done
+
+    while true; do
+        # check Zen operator CSV status
+        csv_status=$(oc get csv ${currentCSV} -n ${csNS} -o jsonpath={.status.phase})
+        if [[ $csv_status == "Succeeded" ]]; then
+            success "Zen operator csv ${currentCSV} status is ${csv_status}."
+            break
+        fi
+        sleep 10
+    done
+}
+
+function zensvc_check() {
+    local csNS=$1
```

Comment:
Usually ZenService is created in Cloud Pak namespace.

Also, we should consider the case that user is trying to run the script with `-a` parameter. It indicates that we will upgrade foundational services in entire cluster, and Cloud Pak namespace is not explicitly specified.